src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/service/AlertCodeService.kt

```diff
@@ -53,4 +54,13 @@ class AlertCodeService(
       }
     }
   }
+
+  fun retrieveAlertCode(alertCode: String): AlertCode =
+    alertCode.let {
+      it.checkAlertCodeExists()
+      alertCodeRepository.findByCode(it)!!.toAlertCodeModel()
```

Comment:
does spring automagically cache value and optimise this, or are we making query to the database 2 times here?

Reply:
ok looked it up myself.. restcontroller funcs are transactional by default, hence query results should be automagically cached and reused : )

Reply:
😄